src/cms/views/push_notifications/push_notification_list_view.py

```diff
@@ -74,9 +75,29 @@ def get(self, request, *args, **kwargs):
                     "region_slug": region.slug,
                 }
             )
-        push_notifications = region.push_notifications.all()
+
+        push_notifications = None
+        search_data = kwargs.get("search_data")
+        if search_data:
+            search_form = ObjectSearchForm(search_data)
+            if search_form.is_valid():
+                query = search_form.cleaned_data["query"]
+                if query:
+                    push_notifications = [
+                        push_translation.push_notification
+                        for push_translation in PushNotificationTranslation.search(
+                            region, language_slug, query
+                        ).order_by("push_notification__created_date")
+                    ]
+
+        if push_notifications is None:
+            query = None
+            push_notifications = list(
+                region.push_notifications.all().order_by("created_date")
+            )
```

Comment:
Right now, the search is only invalid if the length of the query is 0. Wouldn't it make sense in this case to not filter at all?

Reply:
Ah okay. Maybe you could do something like this?
```
search_form = ObjectSearchForm(kwargs.get("search_data"))
if search_form.is_valid():
    ...
else:
   ...
```
(initializing the `ObjectSearchForm` even if the `search_data` is `None` has a little bit of overhead, but it shouldn't matter much)

Another approach might be to perform the filtering on an existing queryset instead of the class `PushNotificationTranslation`, but I'm afraid that's a bit too much work for now...

Reply:
Since queries only get evaluated when they are used, are there any downsides with this approach?
```python
push_notifications = region.push_notifications.all()
query = None

search_data = kwargs.get("search_data")
if search_data:
    search_form = ObjectSearchForm(search_data)
    if search_form.is_valid():
        query = search_form.cleaned_data["query"]
        if query:
            push_notification_keys = set(
                push_translation.push_notification.pk
                for push_translation in PushNotificationTranslation.search(
                    region, language_slug, query
                )
            )
            push_notifications = push_notifications.filter(pk__in=push_notification_keys)
```
This has the upside to also work if at some point other filters specific to the view get implemented, like at the event/page/poi views